lib/trino-parquet/src/main/java/io/trino/parquet/reader/ColumnReaderFactory.java

```diff
@@ -167,7 +167,12 @@ public static ColumnReader create(PrimitiveField field, DateTimeZone timeZone, b
 
         return switch (primitiveType) {
             case BOOLEAN -> new BooleanColumnReader(field);
-            case INT32 -> createDecimalColumnReader(field).orElse(new IntColumnReader(field));
+            case INT32 -> {
+                if (annotation instanceof TimeLogicalTypeAnnotation timeAnnotation && timeAnnotation.getUnit() == MILLIS) {
+                    yield new TimeMillisColumnReader(field);
+                }
+                yield createDecimalColumnReader(field).orElse(new IntColumnReader(field));
```

Comment:
if TimeLogicalTypeAnnotation is present but unit != MILLIS, should we still produce decimal or int reader?
or should we fail "unsupported time unit"?

Reply:
We can update this to
```
if (annotation instanceof TimeLogicalTypeAnnotation timeAnnotation) {
     if (timeAnnotation.getUnit() == MILLIS) {
            yield new TimeMillisColumnReader(field);
     }
     throw unsupportedException(type, field);
}
```
As per the spec, INT32 can be used only for MILLIS unit of TimeLogicalTypeAnnotation